source/core/dictionary.h

```diff
@@ -331,16 +329,34 @@ namespace Slang
 			marks.Clear();
 		}
 
-		template<typename T>
-		bool ContainsKey(const T & key) const
+        TValue* TryGetValueOrAdd(const TKey& key, const TValue& value)
```

Comment:
This is nice, and it would be great to eentually get a variation that takes a lambda to *compute* the value to add, instead of needing the value to be created up front. E.g.:

```c++
auto thing = dict.getOrCreateValue(key, [&]{ return createANewThing(...) });
```

A large fraction of our uses of `Dictionary` are for memo-caches, so there is a lot of code that ends up poking the dictionary twice in the "miss" case so that it can avoid constructing a complex value and then throwing it away.